src/daemon/backends/connman_technology.h

```diff
@@ -0,0 +1,176 @@
+        // ConnMan, the queued value is sent regardless of whether the previous call was successful
+        // or not.
+        //
+        // If a value is received from ConnMan, it is updated directly if there is no pending set.
+        // If a set is pending, the value is updated when the set is finished if there is no other
+        // value queued. The received value may be due to our own set call so "property changed" is
+        // only signalled internally if the value is different, since otherwise it has already been
+        // done for that particular value.
+        //
+        // TODO: Make the above documentation more clear. A bit confusing now but hard. Also, can
+        //       the logic be simplified? Can e.g. pending_/received_ be removed/replaced by a
+        //       booleans and value stored directly in value_ instead of std::optional?
+        // TODO: Add temporary extensive logging and make sure behavior is correct.
+        template <typename V>
+        struct SettableProperty : public sigc::trackable
+        {
+        public:
+            SettableProperty(ConnManTechnology &technology,
+                             PropertyId id,
+                             const Glib::ustring &name,
+                             const PropertyMap &property_map,
+                             const V &default_value);
+
+            const V &value() const;
+            void set(const V &new_value);
+            void changed(const Glib::VariantBase &received_variant);
+
+        private:
+            void set_property(const V &value);
+            void set_property_finish(const Glib::RefPtr<Gio::AsyncResult> &result);
+
+            ConnManTechnology &technology_;
+            const PropertyId id_;
+            const Glib::ustring name_;
+
+            V value_;
+            std::optional<V> pending_;
+            std::optional<V> queued_;
+            std::optional<V> received_;
+        };
+
+        Glib::ustring log_id_str() const;
+
+        void proxy_create_finish(const Glib::RefPtr<Gio::AsyncResult> &result);
+
+        void property_changed(const Glib::ustring &property_name, const Glib::VariantBase &value);
+
+        void scan_finish(const Glib::RefPtr<Gio::AsyncResult> &result);
+
+        Listener &listener_;
+        Glib::RefPtr<Proxy> proxy_;
+
+        const Type type_ = Type::UNKNOWN;
+        const Glib::ustring name_;
+        bool connected_ = false;
+
+        SettableProperty<bool> powered_;
+
+        SettableProperty<bool> tethering_;
+        SettableProperty<Glib::ustring> tethering_identifier_;
+        SettableProperty<Glib::ustring> tethering_passphrase_;
+    };
+
+    // technology_proxy_created() is quaranteed to be called before any other method in Listener.
+    // ConnManBackend will not consider the technology available until this has been done. This is
+    // to make sure that methods can be called and properties can be set on the technology before it
```

Comment:
"can be" -> "can not be" ... this whole sentence is a bit weird... reword... do not want methods to be called before proxy has been created... will crash!